client/src/AppRoot.tsx

```diff
@@ -11,41 +11,59 @@ import CastVoteRecordFiles from './utils/CastVoteRecordFiles'
 import { Storage } from './utils/Storage'
 
 import ElectionManager from './components/ElectionManager'
-import { SaveElection } from './config/types'
+import { SaveElection, SaveCastVoteRecordFiles } from './config/types'
 
 export interface AppStorage {
   election?: Election
+  cvrFiles?: string
 }
 
 export interface Props extends RouteComponentProps {
   storage: Storage<AppStorage>
 }
 
 export const electionStorageKey = 'election'
+export const cvrsStorageKey = 'cvrFiles'
 
 const AppRoot = ({ storage }: Props) => {
+  const printBallotRef = useRef<HTMLDivElement>(null)
+
   const getElection = () => storage.get(electionStorageKey)
+  const getCVRFiles = () => storage.get(cvrsStorageKey)
 
   const storageElection = getElection()
+  const [election, setElection] = useState<OptionalElection>(storageElection)
   const [electionHash, setElectionHash] = useState(
     storageElection ? sha256(JSON.stringify(storageElection)) : ''
   )
-  const [election, setElection] = useState<OptionalElection>(getElection())
-  const printBallotRef = useRef<HTMLDivElement>(null)
 
+  const storageCVRFiles = getCVRFiles()
   const [castVoteRecordFiles, setCastVoteRecordFiles] = useState(
-    CastVoteRecordFiles.empty
+    storageCVRFiles
+      ? CastVoteRecordFiles.empty.import(storageCVRFiles)
+      : CastVoteRecordFiles.empty
   )
 
   const saveElection: SaveElection = (electionDefinition) => {
     setElection(electionDefinition)
     setElectionHash(
       electionDefinition ? sha256(JSON.stringify(electionDefinition)) : ''
     )
-    if (electionDefinition === undefined) {
+    if (electionDefinition) {
+      storage.set(electionStorageKey, electionDefinition)
+    } else {
       storage.remove(electionStorageKey)
+    }
+  }
+
+  const saveCastVoteRecordFiles: SaveCastVoteRecordFiles = (
+    newCastVoteRecordFiles?: CastVoteRecordFiles
+  ) => {
+    setCastVoteRecordFiles(newCastVoteRecordFiles ?? CastVoteRecordFiles.empty)
```

Comment:
This isn't necessarily better, but you could do a default parameter initializer and then in the `if` below check reference equality for `newCastVoteRecordFiles === CastVoteRecordFiles.empty` since empty ones are always the same instance.

Reply:
Yeah, your suggestion seems to read easier.